File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int t) {
        
        Map<Integer, Integer> map = new HashMap<>(); 
        int[] res= new int[2];
        for(int i=0;i<nums.length;i++){
            int rem= t-nums[i]; 
            if(map.containsKey(rem)){
                res[0]=i; 
                res[1]=map.get(rem);
            }
            else map.put(nums[i],i);
        }
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, t):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict={}
        res=[]
        

        for i in range (0, len(nums)): 
            rem= t-nums[i]
            if rem in dict: 
                return [dict[rem],i ] 
                
            dict[nums[i]]= i 
        
        return res